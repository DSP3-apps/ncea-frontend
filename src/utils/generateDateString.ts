import { isEmpty } from './isEmpty';
import { IDateObject } from '../interfaces/searchPayload.interface';

const getMonth = (dateObject: IDateObject, isToDate: boolean): number => {
  let month: number;
  if (isToDate && (isEmpty(dateObject.month) || isNaN(dateObject.month!))) {
    month = 12;
  } else {
    month = !isEmpty(dateObject.month) && !isNaN(dateObject.month!) ? dateObject.month! : 1;
  }
  return month;
};

const generateDateString = (dateObject: IDateObject, isToDate: boolean = false): string => {
  const currentDate = new Date();
  const month = getMonth(dateObject, isToDate);
  let day: number;
  if (isToDate && (isEmpty(dateObject.day) || isNaN(dateObject.day!))) {
    if (month === currentDate.getMonth() + 1 && dateObject.year === currentDate.getFullYear()) {
      day = currentDate.getDate();
    } else {
      day = new Date(Date.UTC(dateObject.year, month, 0)).getDate();
    }
  } else {
    day = !isEmpty(dateObject.day) && !isNaN(dateObject.day!) ? dateObject.day! : 1;
  }

  const date = new Date(Date.UTC(dateObject.year, month - 1, day));

  const dateString = date.toISOString().slice(0, 10);

  return dateString;
};

export { generateDateString };
