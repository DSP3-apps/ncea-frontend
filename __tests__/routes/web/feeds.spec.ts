/**
 * @jest-environment jsdom
 */

'use strict';

import { Server, ServerInjectResponse } from '@hapi/hapi';

import { initializeServer } from '../../../src/infrastructure/server';
import { BASE_PATH } from '../../../src/utils/constants';

jest.mock('../../../src/utils/keyvault', () => ({
  getSecret: jest.fn(),
}));

describe('Feeds Screen', () => {
  let server: Server;
  let response: ServerInjectResponse<object>;
  let document: Document;

  beforeAll((done) => {
    initializeServer().then(async (s: Server) => {
      server = s;

      const options = {
        method: 'GET',
        url: `${BASE_PATH}/feeds`,
      };

      response = await server.inject(options);
      const rawHTML = response.payload;
      const parser = new DOMParser();
      document = parser.parseFromString(rawHTML, 'text/html');
      done();
    });
  });

  afterAll((done) => {
    server.stop().then(() => done());
  });

  describe('Feeds > Sanpshot verification', () => {
    it('should match the Feeds screen snapshot', async () => {
      expect(response.payload).toMatchSnapshot();
    });
  });

  describe('Feeds > Heading', () => {
    it('should render the Feeds content heading', async () => {
      expect(document?.querySelectorAll('.govuk-heading-m')?.[1]?.textContent?.trim()).toBe(
        'NCEA programme – Environment',
      );
      expect(document?.querySelectorAll('.govuk-heading-m')?.[2]?.textContent?.trim()).toBe(
        'Natural Capital – Natural England',
      );
    });
  });
});
