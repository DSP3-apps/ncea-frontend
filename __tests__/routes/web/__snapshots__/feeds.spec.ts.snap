// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feeds Screen Feeds > Sanpshot verification should match the Feeds screen snapshot 1`] = `
"<!DOCTYPE html>
<html lang="en" class="govuk-template app-html-class">
  <head>
    <meta charset="utf-8">
    <title>
  Latest news - Natural Capital Ecosystem Assessment - NCEA â€“ programme
</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, viewport-fit=cover">
    <meta name="theme-color" content="#0b0c0c">

    
  
      <link rel="icon" sizes="48x48" href="/natural-capital-ecosystem-assessment/assets/images/favicon.ico">
      <link rel="icon" sizes="any" href="/natural-capital-ecosystem-assessment/assets/images/favicon.svg" type="image/svg+xml">
      <link rel="mask-icon" href="/natural-capital-ecosystem-assessment/assets/images/govuk-icon-mask.svg" color="#0b0c0c">
      <link rel="apple-touch-icon" href="/natural-capital-ecosystem-assessment/assets/images/govuk-icon-180.png">
      <link rel="manifest" href="/natural-capital-ecosystem-assessment/assets/manifest.json">
    


    
  <link rel="stylesheet" href="/natural-capital-ecosystem-assessment/assets/css/application.css">
  <meta http-equiv="Cache-control" content="no-cache, no-store, must-revalidate">
  <meta http-equiv="Pragma" content="no-cache">
  <meta http-equiv="Expires" content="0">

    
  </head>
  <body class="govuk-template__body app-body-class">
    <script>document.body.className += ' js-enabled' + ('noModule' in HTMLScriptElement.prototype ? ' govuk-frontend-supported' : '');</script>
    
  
<div class="govuk-cookie-banner" data-nosnippet role="region" aria-label="Cookies on the find natural capital data service"
    
        
      
    
   id="cookie_banner">
  
  <div class="govuk-cookie-banner__message govuk-width-container"
    
        
      
    
   id="cookie_block" hidden>

    <div class="govuk-grid-row">
      <div class="govuk-grid-column-two-thirds">
        
        <h2 class="govuk-cookie-banner__heading govuk-heading-m">
          Cookies on the find natural capital data service
        </h2>
        
        <div class="govuk-cookie-banner__content">
          
          <p class="govuk-body">We use some essential cookies to make this service work.</p>
          <p class="govuk-body">We'd like to set additional cookies so we can remember your settings, understand how people use the service and make improvements</p>
          
        </div>
      </div>
    </div>

    
    <div class="govuk-button-group">
    
      
      <button type="button" class="govuk-button" data-module="govuk-button"
          
              
            
          
         id="accept_button">
        Accept analytics cookies
      </button>
    
      
      <button type="button" class="govuk-button" data-module="govuk-button"
          
              
            
          
         id="reject_button">
        Reject analytics cookies
      </button>
    
      
      <a class="govuk-link help-link" href="https://environment.data.gov.uk/help/cookies">View cookies</a>
    
    </div>
    

  </div>
  
  <div class="govuk-cookie-banner__message govuk-width-container" role="alert"
    
        
      
    
   id="accept_block" hidden>

    <div class="govuk-grid-row">
      <div class="govuk-grid-column-two-thirds">
        
        <div class="govuk-cookie-banner__content">
          
          <p class="govuk-body">You've accepted additional cookies. You can <a class="govuk-link help-link" href="https://environment.data.gov.uk/help/cookies">change your cookie settings</a> at any time.</p>
          
        </div>
      </div>
    </div>

    
    <div class="govuk-button-group">
    
      
      <button type="button" class="govuk-button hide_button" data-module="govuk-button">
        Hide cookie message
      </button>
    
    </div>
    

  </div>
  
  <div class="govuk-cookie-banner__message govuk-width-container" role="alert"
    
        
      
    
   id="reject_block" hidden>

    <div class="govuk-grid-row">
      <div class="govuk-grid-column-two-thirds">
        
        <div class="govuk-cookie-banner__content">
          
          <p class="govuk-body">You've rejected additional cookies. You can <a class="govuk-link help-link" href="https://environment.data.gov.uk/help/cookies">change your cookie settings</a> at any time.</p>
          
        </div>
      </div>
    </div>

    
    <div class="govuk-button-group">
    
      
      <button type="button" class="govuk-button hide_button" data-module="govuk-button">
        Hide cookie message
      </button>
    
    </div>
    

  </div>
  
</div>



    
      <a href="#main-content" class="govuk-skip-link" data-module="govuk-skip-link">Skip to main content</a>

    

    
  <script>
    var gtmIdValue = "";
    var keyboardFiltersBaseUrlValue = ""
  </script>
  <script type="module" src="/natural-capital-ecosystem-assessment/assets/scripts/cookieScript.js"></script>
  <header class="header-container">

    <!--Will be removed once Defra GA accounts been created -->

    <!-- Google tag (gtag.js) -->
    

    

    
    
      
    

    

    

    

    <header class="govuk-header govuk-header--full-width-border" data-module="govuk-header">
  <div class="govuk-header__container govuk-width-container">
    <div class="govuk-header__logo">
      <a href="https://www.gov.uk/" class="govuk-header__link govuk-header__link--homepage">
        <svg
          focusable="false"
          role="img"
          class="govuk-header__logotype"
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 148 30"
          height="30"
          width="148"
          aria-label="GOV.UK"
        >
          <title>GOV.UK</title>
          <path d="M22.6 10.4c-1 .4-2-.1-2.4-1-.4-.9.1-2 1-2.4.9-.4 2 .1 2.4 1s-.1 2-1 2.4m-5.9 6.7c-.9.4-2-.1-2.4-1-.4-.9.1-2 1-2.4.9-.4 2 .1 2.4 1s-.1 2-1 2.4m10.8-3.7c-1 .4-2-.1-2.4-1-.4-.9.1-2 1-2.4.9-.4 2 .1 2.4 1s0 2-1 2.4m3.3 4.8c-1 .4-2-.1-2.4-1-.4-.9.1-2 1-2.4.9-.4 2 .1 2.4 1s-.1 2-1 2.4M17 4.7l2.3 1.2V2.5l-2.3.7-.2-.2.9-3h-3.4l.9 3-.2.2c-.1.1-2.3-.7-2.3-.7v3.4L15 4.7c.1.1.1.2.2.2l-1.3 4c-.1.2-.1.4-.1.6 0 1.1.8 2 1.9 2.2h.7c1-.2 1.9-1.1 1.9-2.1 0-.2 0-.4-.1-.6l-1.3-4c-.1-.2 0-.2.1-.3m-7.6 5.7c.9.4 2-.1 2.4-1 .4-.9-.1-2-1-2.4-.9-.4-2 .1-2.4 1s0 2 1 2.4m-5 3c.9.4 2-.1 2.4-1 .4-.9-.1-2-1-2.4-.9-.4-2 .1-2.4 1s.1 2 1 2.4m-3.2 4.8c.9.4 2-.1 2.4-1 .4-.9-.1-2-1-2.4-.9-.4-2 .1-2.4 1s0 2 1 2.4m14.8 11c4.4 0 8.6.3 12.3.8 1.1-4.5 2.4-7 3.7-8.8l-2.5-.9c.2 1.3.3 1.9 0 2.7-.4-.4-.8-1.1-1.1-2.3l-1.2 4c.7-.5 1.3-.8 2-.9-1.1 2.5-2.6 3.1-3.5 3-1.1-.2-1.7-1.2-1.5-2.1.3-1.2 1.5-1.5 2.1-.1 1.1-2.3-.8-3-2-2.3 1.9-1.9 2.1-3.5.6-5.6-2.1 1.6-2.1 3.2-1.2 5.5-1.2-1.4-3.2-.6-2.5 1.6.9-1.4 2.1-.5 1.9.8-.2 1.1-1.7 2.1-3.5 1.9-2.7-.2-2.9-2.1-2.9-3.6.7-.1 1.9.5 2.9 1.9l.4-4.3c-1.1 1.1-2.1 1.4-3.2 1.4.4-1.2 2.1-3 2.1-3h-5.4s1.7 1.9 2.1 3c-1.1 0-2.1-.2-3.2-1.4l.4 4.3c1-1.4 2.2-2 2.9-1.9-.1 1.5-.2 3.4-2.9 3.6-1.9.2-3.4-.8-3.5-1.9-.2-1.3 1-2.2 1.9-.8.7-2.3-1.2-3-2.5-1.6.9-2.2.9-3.9-1.2-5.5-1.5 2-1.3 3.7.6 5.6-1.2-.7-3.1 0-2 2.3.6-1.4 1.8-1.1 2.1.1.2.9-.3 1.9-1.5 2.1-.9.2-2.4-.5-3.5-3 .6 0 1.2.3 2 .9l-1.2-4c-.3 1.1-.7 1.9-1.1 2.3-.3-.8-.2-1.4 0-2.7l-2.9.9C1.3 23 2.6 25.5 3.7 30c3.7-.5 7.9-.8 12.3-.8m28.3-11.6c0 .9.1 1.7.3 2.5.2.8.6 1.5 1 2.2.5.6 1 1.1 1.7 1.5.7.4 1.5.6 2.5.6.9 0 1.7-.1 2.3-.4s1.1-.7 1.5-1.1c.4-.4.6-.9.8-1.5.1-.5.2-1 .2-1.5v-.2h-5.3v-3.2h9.4V28H55v-2.5c-.3.4-.6.8-1 1.1-.4.3-.8.6-1.3.9-.5.2-1 .4-1.6.6s-1.2.2-1.8.2c-1.5 0-2.9-.3-4-.8-1.2-.6-2.2-1.3-3-2.3-.8-1-1.4-2.1-1.8-3.4-.3-1.4-.5-2.8-.5-4.3s.2-2.9.7-4.2c.5-1.3 1.1-2.4 2-3.4.9-1 1.9-1.7 3.1-2.3 1.2-.6 2.6-.8 4.1-.8 1 0 1.9.1 2.8.3.9.2 1.7.6 2.4 1s1.4.9 1.9 1.5c.6.6 1 1.3 1.4 2l-3.7 2.1c-.2-.4-.5-.9-.8-1.2-.3-.4-.6-.7-1-1-.4-.3-.8-.5-1.3-.7-.5-.2-1.1-.2-1.7-.2-1 0-1.8.2-2.5.6-.7.4-1.3.9-1.7 1.5-.5.6-.8 1.4-1 2.2-.3.8-.4 1.9-.4 2.7zM71.5 6.8c1.5 0 2.9.3 4.2.8 1.2.6 2.3 1.3 3.1 2.3.9 1 1.5 2.1 2 3.4s.7 2.7.7 4.2-.2 2.9-.7 4.2c-.4 1.3-1.1 2.4-2 3.4-.9 1-1.9 1.7-3.1 2.3-1.2.6-2.6.8-4.2.8s-2.9-.3-4.2-.8c-1.2-.6-2.3-1.3-3.1-2.3-.9-1-1.5-2.1-2-3.4-.4-1.3-.7-2.7-.7-4.2s.2-2.9.7-4.2c.4-1.3 1.1-2.4 2-3.4.9-1 1.9-1.7 3.1-2.3 1.2-.5 2.6-.8 4.2-.8zm0 17.6c.9 0 1.7-.2 2.4-.5s1.3-.8 1.7-1.4c.5-.6.8-1.3 1.1-2.2.2-.8.4-1.7.4-2.7v-.1c0-1-.1-1.9-.4-2.7-.2-.8-.6-1.6-1.1-2.2-.5-.6-1.1-1.1-1.7-1.4-.7-.3-1.5-.5-2.4-.5s-1.7.2-2.4.5-1.3.8-1.7 1.4c-.5.6-.8 1.3-1.1 2.2-.2.8-.4 1.7-.4 2.7v.1c0 1 .1 1.9.4 2.7.2.8.6 1.6 1.1 2.2.5.6 1.1 1.1 1.7 1.4.6.3 1.4.5 2.4.5zM88.9 28 83 7h4.7l4 15.7h.1l4-15.7h4.7l-5.9 21h-5.7zm28.8-3.6c.6 0 1.2-.1 1.7-.3.5-.2 1-.4 1.4-.8.4-.4.7-.8.9-1.4.2-.6.3-1.2.3-2v-13h4.1v13.6c0 1.2-.2 2.2-.6 3.1s-1 1.7-1.8 2.4c-.7.7-1.6 1.2-2.7 1.5-1 .4-2.2.5-3.4.5-1.2 0-2.4-.2-3.4-.5-1-.4-1.9-.9-2.7-1.5-.8-.7-1.3-1.5-1.8-2.4-.4-.9-.6-2-.6-3.1V6.9h4.2v13c0 .8.1 1.4.3 2 .2.6.5 1 .9 1.4.4.4.8.6 1.4.8.6.2 1.1.3 1.8.3zm13-17.4h4.2v9.1l7.4-9.1h5.2l-7.2 8.4L148 28h-4.9l-5.5-9.4-2.7 3V28h-4.2V7zm-27.6 16.1c-1.5 0-2.7 1.2-2.7 2.7s1.2 2.7 2.7 2.7 2.7-1.2 2.7-2.7-1.2-2.7-2.7-2.7z"></path>
        </svg>
        
      </a>
    </div>
  
    <div class="govuk-header__content">
    
      
      <a href="/natural-capital-ecosystem-assessment" class="govuk-header__link govuk-header__service-name">
        Find natural capital data
      </a>
      
    
    
      <nav aria-label="User" class="govuk-header__navigation">
        <button type="button" class="govuk-header__menu-button govuk-js-header-toggle" aria-controls="navigation" hidden>
          User
        </button>

        <ul id="navigation" class="govuk-header__navigation-list">
        
          
          <li class="govuk-header__navigation-item">
            
              <span>Welcome, Guest&nbsp; <a href='/login?redirect_uri=http%3A%2F%2F0.0.0.0%3A4000%2Fnatural-capital-ecosystem-assessment%2Fnewsfeed' class='govuk-link'>Login</a></span>
            
          </li>
          
        
        </ul>
      </nav>
    
    </div>
  
  </div>
</header>

    

  <div 
class="govuk-service-navigation"
data-module="govuk-service-navigation"
>
    
  <div class="govuk-width-container">

    <div class="govuk-service-navigation__container">
      
      

      
      
        <nav aria-label="Menu" class="govuk-service-navigation__wrapper">
          <button type="button" class="govuk-service-navigation__toggle govuk-js-service-navigation-toggle" aria-controls="secondaryMenu" hidden>
            Menu
          </button>

          <ul class="govuk-service-navigation__list" id="secondaryMenu" >

            
              

              
              <li class="govuk-service-navigation__item">
                
                  <a class="govuk-service-navigation__link" href="/natural-capital-ecosystem-assessment">
                    
                
                Home
              
                  </a>
                
              </li>
            
              

              
              <li class="govuk-service-navigation__item">
                
                  <a class="govuk-service-navigation__link" href="/natural-capital-ecosystem-assessment/about">
                    
                
                About
              
                  </a>
                
              </li>
            
              

              
              <li class="govuk-service-navigation__item">
                
                  <a class="govuk-service-navigation__link" href="/natural-capital-ecosystem-assessment/newsfeed">
                    
                
                News Feed
              
                  </a>
                
              </li>
            

            </ul>
        </nav>
      
    </div>

    </div>

  </div>


  </header>


    
  
      <div class="govuk-width-container">
        
  
<div class="govuk-phase-banner">
  <p class="govuk-phase-banner__content">
    <strong class="govuk-tag govuk-phase-banner__content__tag">
      Beta
    </strong>
    <span class="govuk-phase-banner__text">
      This is a new service - your <a class="govuk-link" rel="noreferrer noopener" target="_blank" href="https://defragroup.eu.qualtrics.com/jfe/form/SV_02KhhVirzVmY0QK">feedback</a> will help us to improve it.
    </span>
  </p>
</div>


        <main class="govuk-main-wrapper app-main-class" id="main-content">
          
  <h1 class="govuk-heading-l">Latest news</h1>
  
  
    <div>
    
      <h2 class="govuk-heading-m">Defra</h2>
      <ul class="govuk-list">
        
          <li class="govuk-width-container feed-container">
            <h3 class="govuk-heading-s">Living England: a national habitat map for everyoneÂ </h3>
            <div class="captions-section">
              <small><b>Author: </b>Amy Woodget</small>
              <small><b>Published on : </b>18 December 2024</small>
            </div>
            <p class="govuk-body">Amy Woodget is the Communications and Engagement Lead for the Living England project. In this post, she introduces the latest release of Natural Englandâ€™s â€˜Living Englandâ€™ dataset â€“ a digital map providing information on Englandâ€™s broad habitats.</p>
            <a class="govuk-link" href="https://defraenvironment.blog.gov.uk/2024/12/18/living-england-a-national-habitat-map-for-everyone/" target="_blank">Living England: a national habitat map for everyoneÂ </a>
            <hr class="govuk-section-break govuk-section-break--l govuk-section-break--visible">
          </li>
        
          <li class="govuk-width-container feed-container">
            <h3 class="govuk-heading-s">Why monitoring matters: showcasing the Natural Capital and Ecosystem Assessment programme</h3>
            <div class="captions-section">
              <small><b>Author: </b>David Jones</small>
              <small><b>Published on : </b>16 October 2024</small>
            </div>
            <p class="govuk-body">In this post, David Jones, Senior Responsible Owner for the terrestrial arm of Defraâ€™s largest research and development programme - the NCEA, shares reflections from the recent stakeholder showcase event.</p>
            <a class="govuk-link" href="https://defraenvironment.blog.gov.uk/2024/10/16/why-monitoring-matters-showcasing-the-natural-capital-and-ecosystem-assessment-programme/" target="_blank">Why monitoring matters: showcasing the Natural Capital and Ecosystem Assessment programme</a>
            <hr class="govuk-section-break govuk-section-break--l govuk-section-break--visible">
          </li>
        
          <li class="govuk-width-container feed-container">
            <h3 class="govuk-heading-s">Mapping our natural assets: the Natural Capital and Ecosystem Assessment programme</h3>
            <div class="captions-section">
              <small><b>Author: </b>Sarah Adcock</small>
              <small><b>Published on : </b>28 March 2024</small>
            </div>
            <p class="govuk-body">Sarah Adcock and David Jones, Senior Responsible Owners for Defra&#39;s Natural Capital and Ecosystem Assessment programme explain why building a picture of our natural assets and their benefits is the first, crucial step in protecting our environment.</p>
            <a class="govuk-link" href="https://defraenvironment.blog.gov.uk/2024/03/28/mapping-our-natural-assets-the-natural-capital-and-ecosystem-assessment-programme/" target="_blank">Mapping our natural assets: the Natural Capital and Ecosystem Assessment programme</a>
            <hr class="govuk-section-break govuk-section-break--l govuk-section-break--visible">
          </li>
        
      </ul>
    </div>


  
    <div>
    
      <h2 class="govuk-heading-m">Natural England</h2>
      <ul class="govuk-list">
        
          <li class="govuk-width-container feed-container">
            <h3 class="govuk-heading-s">The North can be a powerhouse for Nature and growth</h3>
            <div class="captions-section">
              <small><b>Author: </b>Natural England</small>
              <small><b>Published on : </b>07 March 2025</small>
            </div>
            <p class="govuk-body">&quot;Nature is vital to our economy our health and our security. The North has the natural assets that can help create the places people want to live, work and invest in and that will keep them happy and healthy, as â€¦</p>
            <a class="govuk-link" href="https://naturalengland.blog.gov.uk/2025/03/07/the-north-can-be-a-powerhouse-for-nature-and-growth/" target="_blank">The North can be a powerhouse for Nature and growth</a>
            <hr class="govuk-section-break govuk-section-break--l govuk-section-break--visible">
          </li>
        
          <li class="govuk-width-container feed-container">
            <h3 class="govuk-heading-s">How new technologies are helping us monitor one of Englandâ€™s largest grey seal coloniesÂ </h3>
            <div class="captions-section">
              <small><b>Author: </b>Natural England</small>
              <small><b>Published on : </b>26 February 2025</small>
            </div>
            <p class="govuk-body">By Gabriella Fasoli and Nick Tomline (AI specialists), Emma Milner (marine mammal specialist) and Elizabeth Mitchell (engagement), all on the marine Natural Capital and Ecosystem Assessment (mNCEA) programme Documenting Norfolk&#39;s biggest beach party On a windy day in December, something â€¦</p>
            <a class="govuk-link" href="https://naturalengland.blog.gov.uk/2025/02/26/how-new-technologies-are-helping-us-monitor-one-of-englands-largest-grey-seal-colonies/" target="_blank">How new technologies are helping us monitor one of Englandâ€™s largest grey seal coloniesÂ </a>
            <hr class="govuk-section-break govuk-section-break--l govuk-section-break--visible">
          </li>
        
          <li class="govuk-width-container feed-container">
            <h3 class="govuk-heading-s">How can Nature-Based Solutions address the climate and Nature crises?</h3>
            <div class="captions-section">
              <small><b>Author: </b>Tony Juniper</small>
              <small><b>Published on : </b>11 February 2025</small>
            </div>
            <p class="govuk-body">Keynote speech by Tony Juniper CBE, Chair of Natural England, at Nature Returns conference, 6 February 2025 It is an absolute pleasure to be here today and to see this latest journey in what has been a decades-long process of â€¦</p>
            <a class="govuk-link" href="https://naturalengland.blog.gov.uk/2025/02/11/how-can-nature-based-solutions-address-the-climate-and-nature-crises/" target="_blank">How can Nature-Based Solutions address the climate and Nature crises?</a>
            <hr class="govuk-section-break govuk-section-break--l govuk-section-break--visible">
          </li>
        
          <li class="govuk-width-container feed-container">
            <h3 class="govuk-heading-s">Three churrs for the heathland bird survey</h3>
            <div class="captions-section">
              <small><b>Author: </b>Alison Giacomelli</small>
              <small><b>Published on : </b>05 February 2025</small>
            </div>
            <p class="govuk-body">By Alison Giacomelli, Senior Specialist â€“ Ornithology, Natural England Heathlands are home to the eerie sound of churring Nightjars, sweet melody of singing Woodlarks and the rattling song of Dartford Warblers. This year teams of volunteers and fieldworkers will be â€¦</p>
            <a class="govuk-link" href="https://naturalengland.blog.gov.uk/2025/02/05/three-churrs-for-the-heathland-bird-survey/" target="_blank">Three churrs for the heathland bird survey</a>
            <hr class="govuk-section-break govuk-section-break--l govuk-section-break--visible">
          </li>
        
          <li class="govuk-width-container feed-container">
            <h3 class="govuk-heading-s">Maximising the miracle and minimising the disaster â€“ putting Nature at the heart of food security</h3>
            <div class="captions-section">
              <small><b>Author: </b>Tony Juniper</small>
              <small><b>Published on : </b>10 January 2025</small>
            </div>
            <p class="govuk-body">I was invited to speak at a historic event this week â€“ the first major joint gathering of attendees at the two Oxford farming conferences: the Oxford Farming Conference and the Oxford Real Farming Conference. After more than a decade â€¦</p>
            <a class="govuk-link" href="https://naturalengland.blog.gov.uk/2025/01/10/maximising-the-miracle-and-minimising-the-disaster-putting-nature-at-the-heart-of-food-security/" target="_blank">Maximising the miracle and minimising the disaster â€“ putting Nature at the heart of food security</a>
            <hr class="govuk-section-break govuk-section-break--l govuk-section-break--visible">
          </li>
        
          <li class="govuk-width-container feed-container">
            <h3 class="govuk-heading-s">Working together for Nature Recovery in Greater ManchesterÂ Â </h3>
            <div class="captions-section">
              <small><b>Author: </b>Marian Spain</small>
              <small><b>Published on : </b>03 December 2024</small>
            </div>
            <p class="govuk-body">The launch of Biodiversity Net Gain (BNG) has been a major achievement this year â€“ this world-leading initiative means that for the first time, development must leave nature measurably improved. I was lucky enough to visit Greater Manchester to speak â€¦</p>
            <a class="govuk-link" href="https://naturalengland.blog.gov.uk/2024/12/03/working-together-for-nature-recovery-in-greater-manchester/" target="_blank">Working together for Nature Recovery in Greater ManchesterÂ Â </a>
            <hr class="govuk-section-break govuk-section-break--l govuk-section-break--visible">
          </li>
        
          <li class="govuk-width-container feed-container">
            <h3 class="govuk-heading-s">From AI to underwater microphones: improving the evidence base for marine mammalsÂ </h3>
            <div class="captions-section">
              <small><b>Author: </b>Samantha Parker</small>
              <small><b>Published on : </b>13 November 2024</small>
            </div>
            <p class="govuk-body">By Samantha Parker and Emma Milner, Natural England marine mammal senior specialistsÂ  Why are seal population numbers changing so fast along our coasts? How are porpoise feeding patterns changing as our seas become increasingly crowded? And how does the health â€¦</p>
            <a class="govuk-link" href="https://naturalengland.blog.gov.uk/2024/11/13/from-ai-to-underwater-microphones-improving-the-evidence-base-for-marine-mammals/" target="_blank">From AI to underwater microphones: improving the evidence base for marine mammalsÂ </a>
            <hr class="govuk-section-break govuk-section-break--l govuk-section-break--visible">
          </li>
        
          <li class="govuk-width-container feed-container">
            <h3 class="govuk-heading-s">International use of England&amp;#039;s biodiversity metric having global impact</h3>
            <div class="captions-section">
              <small><b>Author: </b>Natural England</small>
              <small><b>Published on : </b>08 November 2024</small>
            </div>
            <p class="govuk-body">By Nick White, Principal Adviser on Biodiversity Net Gain, and Steve Panks, Principal Specialist Biodiversity Net Gain and Biodiversity Metrics, at Natural England, explore the global significance of the biodiversity metric and its role in nature recovery across the world.&amp;#160; â€¦</p>
            <a class="govuk-link" href="https://naturalengland.blog.gov.uk/2024/11/08/international-use-of-englands-biodiversity-metric-having-global-impact/" target="_blank">International use of England&amp;#039;s biodiversity metric having global impact</a>
            <hr class="govuk-section-break govuk-section-break--l govuk-section-break--visible">
          </li>
        
          <li class="govuk-width-container feed-container">
            <h3 class="govuk-heading-s">Seabed Saviours - the EU Life Recreation ReMEDIES project comes to an end</h3>
            <div class="captions-section">
              <small><b>Author: </b>Fiona Tibbitt</small>
              <small><b>Published on : </b>01 November 2024</small>
            </div>
            <p class="govuk-body">By Fiona Tibbitt, Project Manager on the EU Life Recreation ReMEDIESÂ  I have had the privilege of being Project Manager on the EU Life Recreation ReMEDIES project for the last one and half years. Iâ€™ll admit it was a daunting â€¦</p>
            <a class="govuk-link" href="https://naturalengland.blog.gov.uk/2024/11/01/seabed-saviours-the-eu-life-recreation-remedies-project-comes-to-an-end/" target="_blank">Seabed Saviours - the EU Life Recreation ReMEDIES project comes to an end</a>
            <hr class="govuk-section-break govuk-section-break--l govuk-section-break--visible">
          </li>
        
          <li class="govuk-width-container feed-container">
            <h3 class="govuk-heading-s">Mapping our green and blue spaces: the green infrastructure mapping project</h3>
            <div class="captions-section">
              <small><b>Author: </b>Natural England</small>
              <small><b>Published on : </b>31 October 2024</small>
            </div>
            <p class="govuk-body">Martin Moss, senior officer for green infrastructure mapping, and Sarah Parrott, higher officer for engagement and impact If you looked at a map of your local area, could you identify the accessible green spaces? What about the blue ones? If â€¦</p>
            <a class="govuk-link" href="https://naturalengland.blog.gov.uk/2024/10/31/mapping-our-green-and-blue-spaces-the-green-infrastructure-mapping-project/" target="_blank">Mapping our green and blue spaces: the green infrastructure mapping project</a>
            <hr class="govuk-section-break govuk-section-break--l govuk-section-break--visible">
          </li>
        
      </ul>
    </div>


  


        </main>
      </div>
    


    
  <footer class="govuk-footer">
  <div class="govuk-width-container">
    
    <div class="govuk-footer__meta">
      <div class="govuk-footer__meta-item govuk-footer__meta-item--grow">
        
        <h2 class="govuk-visually-hidden">Support links</h2>
        
        <ul class="govuk-footer__inline-list">
        
          <li class="govuk-footer__inline-list-item">
            <a class="govuk-footer__link" href="https://environment.data.gov.uk/support/faqs/275810340/275810354">
              Accessibility statement
            </a>
          </li>
        
          <li class="govuk-footer__inline-list-item">
            <a class="govuk-footer__link" href="https://environment.data.gov.uk/help/cookies">
              Cookie Policy
            </a>
          </li>
        
          <li class="govuk-footer__inline-list-item">
            <a class="govuk-footer__link" href="https://environment.data.gov.uk/support/faqs/275811163/275811177">
              Terms and conditions
            </a>
          </li>
        
          <li class="govuk-footer__inline-list-item">
            <a class="govuk-footer__link" href="https://environment.data.gov.uk/support/faqs/275811163/275811229">
              Privacy
            </a>
          </li>
        
          <li class="govuk-footer__inline-list-item">
            <a class="govuk-footer__link" href="https://environment.data.gov.uk/support">
              Support
            </a>
          </li>
        
        </ul>
        
        
        
        <svg
          aria-hidden="true"
          focusable="false"
          class="govuk-footer__licence-logo"
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 483.2 195.7"
          height="17"
          width="41"
        >
          <path
            fill="currentColor"
            d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
          />
        </svg>
        <span class="govuk-footer__licence-description">
        
          All content is available under the
          <a
            class="govuk-footer__link"
            href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
            rel="license"
          >Open Government Licence v3.0</a>, except where otherwise stated
        
        </span>
      </div>
      <div class="govuk-footer__meta-item">
        <a
          class="govuk-footer__link govuk-footer__copyright-logo"
          href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
        >
        
          Â© Crown copyright
        
        </a>
      </div>
    </div>
  </div>
</footer>

  <div id="overlay" class="overlay"></div>


    
  
  <script type="module" src="/natural-capital-ecosystem-assessment/assets/all.js"></script>
  <script type="module">
    import {initAll} from '/natural-capital-ecosystem-assessment/assets/all.js'
    initAll()
  </script>
  <script>
    const appInsightsConnString = "";
  </script>
  <script type="text/javascript" src="/natural-capital-ecosystem-assessment/assets/scripts/appinsights.js"></script>
  <script type="module" src="/natural-capital-ecosystem-assessment/assets/scripts/customScripts.js"></script>

  </body>
</html>
"
`;
